Sources/TensorFlow/Layers/Recurrent.swift

```diff
@@ -324,9 +329,11 @@ public struct RecurrentLayer<Cell: RecurrentLayerCell>: Layer {
   public typealias Output = [Cell.TimeStepOutput]
 
   public var cell: Cell
+  @noDerivative public let backwardDirection: Bool
 
-  public init(_ cell: @autoclosure () -> Cell) {
+  public init(_ cell: @autoclosure () -> Cell, backwardDirection: Bool = false) {
```

Comment:
It seems strange to add a `backwardDirection` parameter to `RecurrentLayer`'s initializer. It seems like this parameter only exists to support `BidirectionalRecurrentLayer` and doesn't have precedent in other neural network libraries, so it's not really a public API that we want to expose to users.

Suggestion: rather than adding the `backwardDirection` stored property and parameter, how about changing `BidirectionalRecurrentLayer.callAsFunction` to apply the backward recurrent layer to `inputs.reversed()` instead?

```swift
    let backwardOutputs = backward(
      inputs.reversed(), initialState: initialStateBackward)
```

This is less efficient in the short-term but allows us to keep the public APIs that we want. Eventually, we can restore efficiency by making `RecurrentLayer.callAsFunction` take an arbitrary `Collection` of `Cell.TimeStepInput` (e.g. [`ReversedCollection<LazySequence<[Cell.TimeStepInput]>>`](https://developer.apple.com/documentation/swift/reversedcollection)) of instead `[Cell.TimeStepInput]`.

Reply:
I followed Keras, and they do have a `go_backwards` option publicly available, look at https://keras.io/api/layers/recurrent_layers/rnn/. But I am not sure there will be another usage except for `BidirectionalRecurrentLayer`.

But of course, I can change it. Do you think it will make more sense?

Reply:
Aha, thanks for sharing the precedent from Keras, that's good to know.

Unless others feel differently, I still think it's better to avoid adding a `backwardDirection` parameter since it's equivalent to reversing the input and seems ad-hoc.

Could you please pursue the design above instead, which avoids expanding the API surface? If there are strong use cases for a `backwardDirection` parameter outside of `BidirectionalRecurrentLayer`, it can always be added later.

Reply:
Of course, I will look into it later today or tomorrow and change it as you suggested.

Reply:
I have implemented the changes, unfortunately,`.reversed()` is not yet differentiable, so I made a workaround. In the next PRs, I could implement differentiation for it, and they fix it here, what do you think?